Tests/CsDebugScript.Tests/DebugControlTest.cs

```diff
@@ -104,76 +158,122 @@ static void GoBreakContinousVariablesChangeBody()
 
                 int depthOfMainThread = mainThread.StackTrace.Frames.Where(frame => frame.FunctionName.Contains("InfiniteRecursionTestCase")).Count();
 
-                Diagnostics.Debug.WriteLine($"Depth of main thread is {depthOfMainThread}");
                 Assert.Equal(depthOfMainThread, lastArgument + 1);
             }
-        }
+        }, DefaultTimeout);
 
         /// <summary>
-        /// Tests running multiple processes.
+        /// Test that verifies that set breakpoint gets hit.
         /// </summary>
-        static void GoBreakMultipleProcessesBody()
+        [Fact]
+        public void BreakpointSanityTest() => ContinousTestExecutionWrapper(() =>
         {
             InitializeProcess(TestProcessPath, ProcessArguments, DefaultSymbolPath);
+
+            Module module = Process.Current.ModulesByName[TargetModuleName];
+            Variable funcForBreakpoint = module.GetVariable($"{TargetModuleName}!InfiniteRecursionTestCase");
+            BreakpointSpec bpSpec = new BreakpointSpec(funcForBreakpoint.Address);
+
+            var bp = Debugger.AddBreakpoint(bpSpec);
+
+            Debugger.ContinueExecution();
+            bp.WaitForHit();
+        }, DefaultTimeout);
+
+        /// <summary>
+        /// Test that verifies that set breakpoint gets hit with break after hit.
+        /// </summary>
+        [Fact]
+        public void BreakpointBreakAndContinue() => ContinousTestExecutionWrapper(() =>
+        {
             InitializeProcess(TestProcessPath, ProcessArguments, DefaultSymbolPath);
 
-            var ps = CsDebugScript.Process.All;
-            Context.Debugger.ContinueExecution(ps[1]);
-            Context.Debugger.ContinueExecution(ps[0]);
+            Module module = Process.Current.ModulesByName[TargetModuleName];
+            Variable funcForBreakpoint = module.GetVariable($"{TargetModuleName}!InfiniteRecursionTestCase");
```

Comment:
>module.GetVariable($"{TargetModuleName}!InfiniteRecursionTestCase"); [](start = 41, length = 68)

This most probably works only for DbgEng.dll. Since these tests are only targeting that engine, it is probably ok to leave it like this, but you should probably add comment in the code explaining it.

Reply:
Is there any special reason why we don't want to enable similar interface for other debuggers? Exposure of these interfaces don't have anything to do with dbgeng, so is the problem that other debuggers don't support these interface (but they should), or the plan is to deprecate it? What kind of comment do you want me to add?